README.md

```diff
@@ -93,6 +93,57 @@ defmodule ExampleImpl do
   use CarReq, datadog_service_name: :dont_put_me_in_a_box
 end
 ```
+
+### Resource Name Overrides
+
+In most cases, your application's instrumentation modules (the ones that call `:telemetry.attach/4` on Telemetry events executed by CarReq) will take the metadata provided by CarReq's Telemetry events and form a resource name that may look something like:
+
+```
+get some_partner.api.com/product/4123/details
+```
+
+In some cases, you may want specific requests or clients to override the default resource name provided by your application's instrumentation module. To do so, you can pass along a `:resource_name_override` option at the client level or on a per-request
+basis. The value of `:resource_name_override` must be a 1-arity function that takes a string (the default resource_name provided by your
+instrumentation) and returns a new string (the overridden resource_name). Note that when passing this option at the client-level, you
+must pass the option as an external function capture, i.e., `&Module.function/1`. An example would be:
+
+```elixir
+defmodule MyResourceOverrideClient do
+  use CarReq, resource_name_override: &MyResourceOverrideClient.override_function/1
+
+  @doc """
+  Replaces any consecutive instances of numbers in a string with the
+  placeholder `{guid}`. For example
+
+  MyResourceOverrideClient.override_function("get partner.api.com/product/1234/details")
+  # => "get partner.api.com/product/{guid}/details"
```

Comment:
In case you didn't know, sometime ago we added support for `{x}` in [`put_path_params`](https://hexdocs.pm/req/Req.Steps.html#put_path_params/1) e.g.:

```elixir
Req.get!(
  "https://httpbin.org/status/{code}",
  path_params_style: :curly,
  path_params: [code: 201]
).status
#=> 201
```

So if `{x}` is the main use case for this perhaps and you're not using it yet, perhaps worth looking into `path_params` feature? A word of warning, there is a limitation with path params and query strings, https://github.com/wojtekmach/req/issues/424, which could come up when integrating with some services, though.

If you dont' extra flexibility perhaps sticking to `:x`/`{x}` would work but otherwise, yeah, a function with overrides sounds good to me.